forge-ai/src/main/java/forge/ai/ability/DamageDealAi.java

```diff
@@ -746,17 +730,17 @@ private boolean damageChoosingTargets(final Player ai, final SpellAbility sa, fi
                 }
                 return false;
             }
-            // TODO: Improve Damage, we shouldn't just target the player just because we can
             if (sa.canTarget(enemy) && sa.canAddMoreTarget()) {
-                if (((phase.is(PhaseType.END_OF_TURN) && phase.getNextTurn().equals(ai))
+                if ((phase.is(PhaseType.END_OF_TURN) && phase.getNextTurn().equals(ai))
                         || (SpellAbilityAi.isSorcerySpeed(sa, ai) && phase.is(PhaseType.MAIN2))
-                        || ("PingAfterAttack".equals(logic) && phase.getPhase().isAfter(PhaseType.COMBAT_DECLARE_ATTACKERS) && phase.isPlayerTurn(ai))
-                        || immediately || shouldTgtP(ai, sa, dmg, noPrevention)) &&
-                        (!avoidTargetP(ai, sa))) {
-                    tcs.add(enemy);
-                    if (divided) {
-                        sa.addDividedAllocation(enemy, dmg);
-                        break;
+                        || immediately) {
+                    boolean pingAfterAttack = "PingAfterAttack".equals(logic) && phase.getPhase().isAfter(PhaseType.COMBAT_DECLARE_ATTACKERS) && phase.isPlayerTurn(ai);
+                    if (pingAfterAttack || shouldTgtP(ai, sa, dmg, noPrevention)) {
```

Comment:
logic should be:
`((pingAfterAttack && !avoidTargetP(ai, sa)) || shouldTgtP(ai, sa, dmg, noPrevention))`

Reply:
Good catch! :) Done